streams/src/test/java/org/apache/kafka/streams/integration/StoresIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.ClassRule;
+import static org.junit.Assert.assertEquals;
+import java.util.Properties;
+
+import org.apache.kafka.streams.integration.utils.EmbeddedSingleNodeKafkaCluster;
+
+
+public class StoresIntegrationTest {
+    @ClassRule
+    public static EmbeddedSingleNodeKafkaCluster cluster = new EmbeddedSingleNodeKafkaCluster();
+    private static final String INPUT_TOPIC = "inputTopic";
+    private static final String OUTPUT_TOPIC = "outputTopic";
+    private static int foundStores = 0;
+
+    @BeforeClass
+    public static void startKafkaCluster() throws Exception {
+        cluster.createTopic(INPUT_TOPIC);
+        cluster.createTopic(OUTPUT_TOPIC);
+    }
+
+    private static class MyProcessorSupplier implements ProcessorSupplier<String, String> {
+        private final int numStores;
+        MyProcessorSupplier(int numStores) {
+            this.numStores = numStores;
+        }
+        @Override
+        public Processor<String, String> get() {
+            return new Processor<String, String>() {
+                private ProcessorContext context;
+
+                @Override
+                @SuppressWarnings("unchecked")
+                public void init(ProcessorContext context) {
+                    this.context = context;
+                    this.context.schedule(1000);
+                    for (int i = 0; i < numStores; i++) {
+                        String storeName = "Counts" + i;
+                        if (context.getStateStore(storeName) != null) {
+                            foundStores++;
+                        }
+                    }
+                }
+
+                @Override
+                public void process(String dummy, String line) {
+                    // do nothing
+                }
+
+                @Override
+                public void punctuate(long timestamp) {
+                    // do noting
+                }
+
+                @Override
+                public void close() {
+                    // do nothing
+                }
+            };
+        }
+    }
+
+
+    /**
+     * This tests the problem described in https://issues.apache.org/jira/browse/KAFKA-3559
+     */
+    @Test
+    public void testCreateLargeNumberOfStores() throws Exception {
+        int numStores = 100;
+        Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "stores-integration-test-processor");
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.bootstrapServers());
+        props.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, cluster.zKConnectString());
+        props.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        props.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+
+        // setting offset reset to earliest so that we can re-run the demo code with the same pre-loaded data
```

Comment:
Was this comment and setting copy-pasted from elsewhere?  Seems a bit out of place, given that (a) this test class creates its own embedded Kafka cluster from scratch and (b) there's no data being pre-loaded as far as I can tell.